app/packages/core/src/Dataset.ts

```diff
@@ -8,13 +8,33 @@ import {
   DatasetQuery$data,
 } from "./__generated__/DatasetQuery.graphql";
 
+export const DatasetSavedViewsFragment = graphql`
+  fragment DatasetSavedViewsFragment on Query
+  @refetchable(queryName: "DatasetSavedViewsFragmentQuery") {
+    savedViews(datasetName: $name) {
+      id
+      datasetId
+      name
+      slug
+      description
+      color
+      viewStages
```

Comment:
Since we don't actually set the view using `viewStages`, we can probably remove this from the fragment to avoid passing around potentially really long stage definitions that are not viewed.